src/Symfony/Component/Cache/Adapter/ArrayAdapter.php

```diff
@@ -111,7 +112,7 @@ public function hasItem($key)
 
             return true;
         }
-        CacheItem::validateKey($key);
+        \assert('' !== CacheItem::validateKey($key));
```

Comment:
Why check for `'' !==`? It does not matter what the method returns as it validates using an exception. It looks like the return value is important here but`\assert(CacheItem::validateKey($key));` would be enough

Reply:
thus, passing something else than a boolean to `assert` has a different behavior. It would treat it as an expression to run.